packages/compass-components/src/hooks/use-hotkeys.ts

```diff
@@ -0,0 +1,38 @@
+import { useMemo } from 'react';
+import { useHotkeys as useGlobalHotkeys } from 'react-hotkeys-hook';
+
+const isMac = () => navigator.userAgent.indexOf('Mac') !== -1;
+
+/**
+ *
+ * Modifier | macOS       | Windows/Linux
+ * -------- | ----------- | -------------
+ * ctrl     | control     | ctrl
+ * shift    | shift       | shift
+ * alt      | option      | alt
+ * meta     | command     | ctrl
+ *
+ */
+export const useHotkeys = (...args: Parameters<typeof useGlobalHotkeys>) => {
+  const [key] = args;
+  const ref = useGlobalHotkeys(...args);
+
+  const shortcut = useMemo(() => {
+    if (Array.isArray(key)) {
+      return key.map(mapKeyToShortcut);
+    }
+    return mapKeyToShortcut(key);
+  }, [key]);
+
+  return {
+    shortcut,
+    ref,
+  };
+};
+
+const mapKeyToShortcut = (key: string) => {
+  if (isMac()) {
+    return key.replace('meta', '⌘').replace('alt', 'option');
```

Comment:
is replacing enough here? one issue that comes to mind is that we may forget or use the spacing inconsistently `ctrl+c` vs `ctrl + c`.

Reply:
this i left intentionally like this and let the user decide how it should look. if they want space, they will add a shortcut with space `ctrl + x`.

Reply:
> the user decides how it should look

I don't see why that would be useful for us. We are the users here and having different users to make shortcuts look differently is what we want to avoid: the main benefit of having this as a shared function is to get consistent results, I believe we should choose a way (with or without spaces), stick to that, and make sure the result is what we expect.

If there is a reason to allow different formats I'd rather have that as an option, so we are sure we don't use it by mistake. Personally I would prefer that we don't even have the option to mess it up though.

Reply:
added [here](https://github.com/mongodb-js/compass/pull/4108/commits/5f00f63af8b075c40aa669209b9d29c4cdf568fc#diff-0ca4f234b23391bba7137f47e9a314b40dfa18d85a79fdeee81a6a71b04d0953R35) in [5f00f63](https://github.com/mongodb-js/compass/pull/4108/commits/5f00f63af8b075c40aa669209b9d29c4cdf568fc)